pkg/restore/restore.go

```diff
@@ -1831,3 +1874,81 @@ func (ctx *restoreContext) getSelectedRestoreableItems(resource, targetNamespace
 	}
 	return restorable, warnings, errs
 }
+
+// removeRestoreLabels removes the restore name and the
+// restored backup's name.
+func removeRestoreLabels(obj metav1.Object) {
+	labels := obj.GetLabels()
+
+	if labels == nil {
+		labels = make(map[string]string)
+	}
+
+	labels[velerov1api.BackupNameLabel] = ""
+	labels[velerov1api.RestoreNameLabel] = ""
+
+	obj.SetLabels(labels)
+}
+
+// updates the backup/restore labels
+func (ctx *restoreContext) updateBackupRestoreLabels(fromCluster, fromClusterWithLabels *unstructured.Unstructured, namespace string, resourceClient client.Dynamic) (warnings, errs Result) {
+	patchBytes, err := generatePatch(fromCluster, fromClusterWithLabels)
+	if err != nil {
+		ctx.log.Infof("error generating patch for %s %s: %v", fromCluster.GroupVersionKind().Kind, kube.NamespaceAndName(fromCluster), err)
+		errs.Add(namespace, err)
+		return warnings, errs
+	}
+
+	if patchBytes == nil {
+		// In-cluster and desired state are the same, so move on to
+		// the next items
+		ctx.log.Infof("skipped updating backup/restore labels for %s %s: in-cluster and desired state are the same along-with the labels", fromCluster.GroupVersionKind().Kind, kube.NamespaceAndName(fromCluster))
+		return warnings, errs
```

Comment:
I think here it should also generate an error and thus an error log. From the 1st parameter "fromCluster" and 2nd parameter "fromClusterWithLabels", this function expects the 1st parameter and 2nd parameter to be different, but if then the generated patch is the same, this should be an unexpected case, and in this case, this function doesn't finish its target actually.